api/app/signals/apps/signals/admin.py

```diff
@@ -206,6 +206,18 @@ class RoutingExpressionAdmin(admin.ModelAdmin):
     list_display = ['_expression', '_department', 'is_active']
 
     def save_model(self, request, obj, form, change):
+        dept = form.cleaned_data.get('_department', None)
+        user = form.cleaned_data.get('_user', None)
+
+        if dept and user and not user.profile.departments.filter(id=dept.id):
+            messages.add_message(
```

Comment:
Yes looks good, please add small unit test. (I did test the code locally, but better to have small unit test to prevent us from breaking stuff in the future)